test/cri-containerd/runpodsandbox_test.go

```diff
@@ -1460,29 +1460,50 @@ func Test_RunPodSandbox_ProcessDump_LCOW(t *testing.T) {
 		},
 	}
 
-	// Wait for the first container to die and create the core dump.
-	time.Sleep(time.Second * 5)
-
 	container2ID := createContainer(t, client, ctx, c2Request)
 	defer removeContainer(t, client, ctx, container2ID)
 
 	startContainer(t, client, ctx, container2ID)
 	defer stopContainer(t, client, ctx, container2ID)
 
-	// Check if the core dump file is present
-	execCommand := []string{
-		"ls",
-		"/coredumps/core",
-	}
-	execRequest := &runtime.ExecSyncRequest{
-		ContainerId: container2ID,
-		Cmd:         execCommand,
-		Timeout:     20,
+	checkForDumpFile := func() error {
+		// Check if the core dump file is present
+		execCommand := []string{
+			"ls",
+			"/coredumps/core",
+		}
+		execRequest := &runtime.ExecSyncRequest{
+			ContainerId: container2ID,
+			Cmd:         execCommand,
+			Timeout:     20,
+		}
+
+		r := execSync(t, client, ctx, execRequest)
+		if r.ExitCode != 0 {
+			return fmt.Errorf("failed with exit code %d running `ls`: %s", r.ExitCode, string(r.Stderr))
+		}
+		return nil
 	}
 
-	r := execSync(t, client, ctx, execRequest)
-	if r.ExitCode != 0 {
-		t.Fatalf("failed with exit code %d running `ls`: %s", r.ExitCode, string(r.Stderr))
+	var (
+		done    bool
+		timeout = time.After(time.Second * 10)
+	)
+	for {
+		if done {
+			break
+		}
```

Comment:
nit: we could use `for !done` here instead

Reply:
Oh good point, I'll change and check in after